Add tests for Git remote URL changes

rcutmore opened the pull request:
##### ISSUE TYPE

<!--- Pick one below and delete the rest: -->
- Bugfix Pull Request
##### ANSIBLE VERSION

<!--- Paste verbatim output from “ansible --version” between quotes below -->

```
ansible 2.2.0 (git-integration-tests 70e58ffcd8) last updated 2016/07/30 11:26:19 (GMT -400)
  lib/ansible/modules/core: (issue-4006 4f493ff7ca) last updated 2016/07/31 13:31:41 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 68ca157f3b) last updated 2016/07/18 07:22:17 (GMT -400)
  config file = 
  configured module search path = Default w/o overrides

```
##### SUMMARY

<!--- Describe the change, including rationale and design decisions -->

This updates existing Git integration tests and adds new tests for changes to the Git module ([PR here](https://github.com/ansible/ansible-modules-core/pull/4243)). Both pull requests should be merged together.

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

rcutmore committed:
Update Git tests for set remote URL changes

rcutmore committed:
Add Git tests for set remote URL changes

robinro commented on test/integration/roles/test_git/tasks/main.yml:
```diff
@@ -435,6 +437,60 @@
+      - not checkout_same_url|changed
+  when: git_version.stdout | version_compare("{{git_version_supporting_get_url}}", '>=')
+
+- name: check repo changed (when get url command not supported in git version)
```
I would not ensure that the bug is there. Only test for desired behavior. Otherwise someone fixing this behavior in the future will make these tests fail.

rcutmore replied:
Got it, will do.

robinro commented on test/integration/roles/test_git/tasks/main.yml:
```diff
@@ -435,6 +437,60 @@
+- name: clone repo with new url in check mode
+  git: repo={{ repo_update_url_1 }} dest={{ checkout_dir }}
+  register: checkout_new_url_check_mode
+  check_mode: True
```
Yeah check_mode!

robinro commented on test/integration/roles/test_git/tasks/main.yml:
```diff
@@ -495,7 +551,7 @@
 - assert:
     that: "not checkout2|changed"
-  when: git_version.stdout | version_compare("{{git_version_supporting_depth}}", '>=')
+  when: git_version.stdout | version_compare("{{git_version_supporting_get_url}}", '>=')
```
Why did you change this? Looks like too much search/replace.

rcutmore replied:
The tests that check for the version fail without the version check. Versions lower than 2.7 will report being changed due to not having `git remote get-url`.

robinro replied:
So for this you need version >-= get_url and version >= depth, i.e. one can revert it, once you go to ls-remote (since 1.9 for depth is bigger than 1.7 for ls-remote)

robinro commented on test/integration/roles/test_git/tasks/main.yml:
```diff
@@ -555,14 +611,19 @@
+  assert:
+    that:
+      - not update_new_tag|changed
+  when: git_version.stdout | version_compare("{{git_version_supporting_get_url}}", '>=')
```
why do you check for version here?

rcutmore commented:
@robinro Thanks for reviewing. I update these tests along with the change to use `git ls-remote --get-url`.

rcutmore committed:
Update Git tests for fetch remote URL change

rcutmore commented:
@robinro I updated the tests and ran them against a newer and older version of Git. From what I could see `ls-remote --get-url` was added in 1.7.5, but I can update this if I'm mistaken.

robinro commented on test/integration/roles/test_git/tasks/main.yml:
```diff
@@ -495,7 +545,7 @@
 - assert:
     that: "not checkout2|changed"
-  when: git_version.stdout | version_compare("{{git_version_supporting_depth}}", '>=')
+  when: git_version.stdout | version_compare("{{git_version_supporting_depth}}", '>=') and git_version.stdout | version_compare("{{git_version_supporting_depth}}", '>=')
```
the second check should be version_geturl

rcutmore replied:
Thanks for catching that, just pushed a fix.

rcutmore committed:
Fix version check in Git integration tests

rcutmore committed:
Remove unnecessary version checks in Git tests

robinro commented:
@jctanner Please consider merging this. You already merged the changes to the git module https://github.com/ansible/ansible-modules-core/pull/4243, which are covered by these tests.

jctanner commented:
@robinro @rcutmore very sorry I missed this. Assigning to myself.

jctanner commented:
@rcutmore github thinks there is a merge conflict although I don't see one locally. Can you rebase and maybe add a new commit so we can try to trigger shippable to retest and github to re-check?

mattclay commented:
The integration test roles have moved, which is why this PR needs to be updated.

rcutmore commented:
@jctanner I rebased this PR but more recent changes have broken the check_mode test when running them locally. It looks like it has something to do with [this commit](https://github.com/ansible/ansible-modules-core/commit/e4f40d1cdda9161849098bfeac9d360fe9dd77f3) because the test passes before that point in time but I haven't had a chance to investigate further yet.

rcutmore committed:
Git: report changed when needed in check mode

rcutmore committed:
Exclude git versions missing ls_remote from test

rcutmore commented:
@jctanner I updated this PR for the recent repo merge and added the required fix that was submitted to the old core modules repo before the merge ([here](https://github.com/ansible/ansible-modules-core/pull/5374)).

rcutmore commented:
ready_for_review

mattclay commented:
@rcutmore Thanks for the fix and the new tests!
